Add test case for async HTTPX client

Failing consistently on only macOS with the following error:

```
______________________ test_async_httpx_works_with_inject ______________________

cls = <class '_pytest.runner.CallInfo'>
func = <function call_runtest_hook.<locals>.<lambda> at 0x11222e320>
when = 'call'
reraise = (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)

    @classmethod
    def from_call(
        cls,
        func: "Callable[[], TResult]",
        when: "Literal['collect', 'setup', 'call', 'teardown']",
        reraise: Optional[
            Union[Type[BaseException], Tuple[Type[BaseException], ...]]
        ] = None,
    ) -> "CallInfo[TResult]":
        """Call func, wrapping the result in a CallInfo.
    
        :param func:
            The function to call. Called without arguments.
        :param when:
            The phase in which the function is called.
        :param reraise:
            Exception or exceptions that shall propagate if raised by the
            function, instead of being wrapped in the CallInfo.
        """
        excinfo = None
        start = timing.time()
        precise_start = timing.perf_counter()
        try:
>           result: Optional[TResult] = func()

.nox/test/lib/python3.10/site-packages/_pytest/runner.py:339: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.nox/test/lib/python3.10/site-packages/_pytest/runner.py:260: in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
.nox/test/lib/python3.10/site-packages/pluggy/_hooks.py:265: in __call__
    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
.nox/test/lib/python3.10/site-packages/pluggy/_manager.py:80: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
.nox/test/lib/python3.10/site-packages/_pytest/unraisableexception.py:88: in pytest_runtest_call
    yield from unraisable_exception_runtest_hook()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def unraisable_exception_runtest_hook() -> Generator[None, None, None]:
        with catch_unraisable_exception() as cm:
            yield
            if cm.unraisable:
                if cm.unraisable.err_msg is not None:
                    err_msg = cm.unraisable.err_msg
                else:
                    err_msg = "Exception ignored in"
                msg = f"{err_msg}: {cm.unraisable.object!r}\n\n"
                msg += "".join(
                    traceback.format_exception(
                        cm.unraisable.exc_type,
                        cm.unraisable.exc_value,
                        cm.unraisable.exc_traceback,
                    )
                )
>               warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
E               pytest.PytestUnraisableExceptionWarning: Exception ignored in: <socket.socket fd=-1, family=AddressFamily.AF_INET6, type=SocketKind.SOCK_STREAM, proto=0>
E               
E               Traceback (most recent call last):
E                 File "/Users/runner/work/truststore/truststore/.nox/test/lib/python3.10/site-packages/anyio/_core/_eventloop.py", line 153, in get_asynclib
E                   return sys.modules[modulename]
E               KeyError: 'anyio._backends._asyncio'
E               
E               During handling of the above exception, another exception occurred:
E               
E               Traceback (most recent call last):
E                 File "/Users/runner/hostedtoolcache/Python/3.10.10/x64/lib/python3.10/ast.py", line 50, in parse
E                   return compile(source, filename, mode, flags,
E               ResourceWarning: unclosed <socket.socket fd=15, family=AddressFamily.AF_INET6, type=SocketKind.SOCK_STREAM, proto=0, laddr=('::1', 9999, 0, 0)>

.nox/test/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning
```

--- a/tests/test_inject.py
+++ b/tests/test_inject.py
@@ -113,9 +113,17 @@ async def test_requests_works_with_inject(server: Server) -> None:
 @pytest.mark.usefixtures("inject_truststore")
 async def test_sync_httpx_works_with_inject(server: Server) -> None:
     def test_httpx():
-        client = httpx.Client()
-        resp = client.request("GET", server.base_url)
-        assert resp.status_code == 200
+        with httpx.Client() as client:
+            resp = client.request("GET", server.base_url)
+            assert resp.status_code == 200
 
     thread = asyncio.to_thread(test_httpx)
     await thread
+
+
+@pytest.mark.usefixtures("inject_truststore")
+@pytest.mark.asyncio
+async def test_async_httpx_works_with_inject(server: Server) -> None:
+    async with httpx.AsyncClient() as client:
+        resp = await client.request("GET", server.base_url)
+        assert resp.status_code == 200

